crates/bevy_math/src/angle/mod.rs

```diff
@@ -0,0 +1,312 @@
+use std::ops::{
+    Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Rem, RemAssign, Sub, SubAssign,
+};
+
+mod degrees;
+mod radians;
+
+pub use degrees::Degrees;
+pub use radians::Radians;
+
+/// A trait for functionality and constants shared by the [`Degrees`] and [`Radians`] types.
+pub trait Angle:
+    Sized
+    + Clone
+    + Copy
+    + Add<Output = Self>
+    + Sub<Output = Self>
+    + Mul<Output = Self>
+    + Div<Output = Self>
+    + Rem<Output = Self>
+    + Neg<Output = Self>
+    + PartialEq
+    + PartialOrd
+{
+    /// An angle of zero.
+    const ZERO: Self;
+    /// A `45` degree angle. Equivalent to `π/4` radians.
+    const EIGHTH: Self;
+    /// A `90` degree angle (right angle). Equivalent to `π/2` radians.
+    #[doc(alias = "RIGHT_ANGLE")]
+    const QUARTER: Self;
+    /// A `180` degree angle (straight angle). Equivalent to `π` radians.
+    #[doc(alias = "STRAIGHT_ANGLE")]
+    const HALF: Self;
+    /// A `360` degree angle (perigon). Equivalent to `2π` radians.
+    #[doc(alias("PERIGON", "FULL_ANGLE"))]
+    const FULL: Self;
+
+    /// Creates a new angle.
+    fn new(angle: f32) -> Self;
+
+    /// Returns the angle value contained within `self`.
+    fn value(self) -> f32;
+
+    /// Returns the sine of the angle.
+    fn sin(self) -> f32;
+
+    /// Returns the cosine of the angle.
+    fn cos(self) -> f32;
+
+    /// Returns the tangent of the angle.
+    fn tan(self) -> f32;
+
+    /// Returns the sine and cosine of the angle.
+    fn sin_cos(self) -> (f32, f32);
+
+    /// Returns the absolute angle.
+    #[inline]
+    fn abs(self) -> Self {
+        Self::new(self.value().abs())
+    }
+
+    /// Returns the minimum of the two angles, ignoring NaN.
+    #[inline]
+    fn min(self, other: Self) -> Self {
+        Self::new(self.value().min(other.value()))
+    }
+
+    /// Returns the maximum of the two angles, ignoring NaN.
+    #[inline]
+    fn max(self, other: Self) -> Self {
+        Self::new(self.value().max(other.value()))
+    }
+
+    /// Normalizes the angle to be within the range of `[0, 360)` degrees or `[0, 2π)` radians.
+    #[inline]
+    fn normalized(self) -> Self {
+        if self < Self::FULL && self > Self::ZERO {
```

Comment:
This contradicts the doc for `0`, which should already be normalized and require no work. Anyone be `>=0`.

Reply:
Thanks, I'll fix that to `>=`